gui/dist_view/viewport_drawing.cpp

```diff
@@ -521,16 +539,34 @@ void Viewport::drawLegend(QPainter *painter, int sel)
 			painter->setPen(Qt::white);
 	}
 
+
+    //drawing background for y-axis
+    QPointF ytp(0, 0);
+    QPointF ybp(60, height);
+    QRectF ybgrect(ytp, ybp);
+
+    painter->save();
+    painter->setPen(QColor(0,0,0, 128));
+    painter->fillRect(ybgrect, QBrush(QColor(0,0,0,128)));
+    painter->drawRect(ybgrect);
+    painter->restore();
+
+
 	/// y-axis
 	for (size_t i = 0; i < yaxis.size(); ++i) {
 		float ifrac = (float)(i)/(float)(yaxis.size()-1) * (float)((*ctx)->nbins - 1);
-		QPointF b = modelview.map(QPointF(0.f, (float)((*ctx)->nbins - 1) - ifrac));
-		b += QPointF(-8.f, 20.f); // draw left of data, vcenter alignment
-		QPointF t = b;
-		t -= QPointF(200.f, 40.f); // draw left of data, vcenter alignment
-		QRectF rect(t, b);
-		painter->drawText(rect, Qt::AlignVCenter | Qt::AlignRight, yaxis[i]);
-	}
+        QPointF b = modelview.map(QPointF(0.f, (float)((*ctx)->nbins - 1) - ifrac));
+
+        QPointF t = b;
+        t -= QPointF(200.f, 40.f); // draw left of data, vcenter alignment
```

Comment:
we set x to zero afterwards. so here only change y and update comment accordingly.